tests/components/vizio/test_media_player.py

```diff
@@ -329,3 +330,28 @@
             blocking=True,
         )
         assert service_call.call_count == 1
+
+
+async def test_options_update(
+    hass: HomeAssistantType, vizio_connect: pytest.fixture, vizio_update: pytest.fixture
+) -> None:
+    """Test for when config entry update event fires."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, data=MOCK_SPEAKER_CONFIG, unique_id=UNIQUE_ID
+    )
+    await _test_init(
+        hass,
+        config_entry,
+        VIZIO_DEVICE_CLASS_SPEAKER,
+        DEVICE_CLASS_SPEAKER,
+        True,
+        STATE_ON,
+    )
+
+    updated_options = {CONF_VOLUME_STEP: VOLUME_STEP}
+    new_options = config_entry.options.copy()
+    new_options.update(updated_options)
+    hass.config_entries.async_update_entry(
```

Comment:
What do we want to test here? We probably want to test that the integration is updated properly after the options have been updated.

Reply:
I added a test that calls the `volume_up` service and asserts that the volume step is set to the new value